plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/ServicesAndRoutingPage.java

```diff
@@ -116,15 +122,37 @@ private void createExposedPortsControl(Composite parent, DataBindingContext dbc)
 			.applyTo(label);
 		Composite tableContainer = new Composite(container, SWT.NONE);
 		
+        IObservableList portsObservable = BeanProperties.list(
+                IServiceAndRoutingPageModel.PROPERTY_SERVICE_PORTS).observe(model);
 		portsViewer = createTable(tableContainer);
+		ObservableListContentProvider contentProvider = new ObservableListContentProvider();
+		portsViewer.setContentProvider(contentProvider);
+		ObservableMapLabelProvider labelProvider = new ObservableMapLabelProvider(Properties.observeEach(contentProvider.getKnownElements(), BeanProperties.values(ServicePortAdapter.NAME, ServicePortAdapter.PORT, ServicePortAdapter.TARGET_PORT, ServicePortAdapter.ROUTE_PORT))) {
```

Comment:
Can you, please, break this line. It's length (280) is too big and uncomfortable to read